pkg/tests/end_to_end_tests/rules_test.go

```diff
@@ -0,0 +1,95 @@
+// LICENSE for a copy of the license.
+
+package end_to_end_tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+
+	"github.com/timescale/promscale/pkg/pgclient"
+	"github.com/timescale/promscale/pkg/pgmodel/cache"
+	"github.com/timescale/promscale/pkg/query"
+	"github.com/timescale/promscale/pkg/rules"
+	"github.com/timescale/promscale/pkg/tenancy"
+)
+
+const RecordingRulesEvalConfigPath = "../testdata/rules/config.recording_rules_eval.yaml"
+
+func TestRecordingRulesEval(t *testing.T) {
+	withDB(t, *testDatabase, func(db *pgxpool.Pool, t testing.TB) {
+		conf := &pgclient.Config{
+			CacheConfig:             cache.DefaultConfig,
+			WriteConnectionsPerProc: 4,
+			MaxConnections:          -1,
+		}
+
+		pgClient, err := pgclient.NewClientWithPool(conf, 1, db, tenancy.NewNoopAuthorizer(), false)
+		require.NoError(t, err)
+		defer pgClient.Close()
+		err = pgClient.InitPromQLEngine(&query.Config{
+			MaxQueryTimeout:      query.DefaultQueryTimeout,
+			EnabledFeatureMap:    map[string]struct{}{"promql-at-modifier": {}},
+			SubQueryStepInterval: query.DefaultSubqueryStepInterval,
+			LookBackDelta:        query.DefaultLookBackDelta,
+			MaxSamples:           query.DefaultMaxSamples,
+			MaxPointsPerTs:       11000,
+		})
+		require.NoError(t, err)
+
+		ingestor := pgClient.Ingestor()
+		ts := generateSmallTimeseries()
+		_, _, err = ingestor.Ingest(context.Background(), newWriteRequestWithTs(copyMetrics(ts)))
+		require.NoError(t, err)
+
+		rulesCfg := &rules.Config{
+			NotificationQueueCapacity: rules.DefaultNotificationQueueCapacity,
+			OutageTolerance:           rules.DefaultOutageTolerance,
+			ForGracePeriod:            rules.DefaultForGracePeriod,
+			ResendDelay:               rules.DefaultResendDelay,
+			PrometheusConfigAddress:   RecordingRulesEvalConfigPath,
+		}
+		require.NoError(t, rules.Validate(rulesCfg))
+		require.True(t, rulesCfg.ContainsRules())
+
+		manager, err := rules.NewManager(context.Background(), prometheus.NewRegistry(), pgClient, rulesCfg)
+		require.NoError(t, err)
+
+		require.NotNil(t, rulesCfg.PrometheusConfig)
+		require.NoError(t, manager.ApplyConfig(rulesCfg.PrometheusConfig))
+		require.NoError(t, manager.Update(time.Millisecond*100, rulesCfg.PrometheusConfig.RuleFiles, nil, ""))
+
+		stopManager := make(chan struct{})
+		go func() {
+			go func() {
+				require.NoError(t, manager.Run(), "error running rules manager")
+			}()
+			<-stopManager
+			manager.Stop()
+		}()
+
+		checkResult := func() {
+			// Check if recording rule as a metric exists in metric catalog table.
+			var exists bool
+			err := db.QueryRow(context.Background(), "select count(*)>0 from _prom_catalog.metric where metric_name = 'test_rule'").Scan(&exists)
+			require.NoError(t, err)
+			require.True(t, exists)
+
+			// Check if the sum is right.
+			expected := 1.0
```

Comment:
I quickly checked `generateSmallTimeseries()` and the samples are: 
```
	{Timestamp: 4, Value: 0.4},
				{Timestamp: 5, Value: 0.5},
```
So I am confused how does `0.4 + 0.5` add to `1.0`?